src/data/Data.js

```diff
@@ -27,12 +27,12 @@ Data.prototype = {
 			// configurability must be true to allow the property to be
 			// deleted with the delete operator
 			} else {
-				descriptor[ this.expando ] = {
+				descriptor = {
 					value: value,
 					writable: true,
 					configurable: true
 				};
-				Object.defineProperties( owner, descriptor );
+				Object.defineProperty( owner, this.expando, descriptor );
```

Comment:
And do we want the `descriptor` variable when it is only assigned and used once? Currently it is also assigned twice but only used in this one spot...